docs/IntegrationTest.md

```diff
@@ -0,0 +1,63 @@
+<!--
+ Copyright 2019 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ compliance with the License. You may obtain a copy of the License at
+
+        https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under the License
+ is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ implied. See the License for the specific language governing permissions and limitations under the
+ License.
+-->
+# Integration Tests
+
+## Jenkins on GKE with helm
+Follow Google Cloud Solutions'
+[tutorial](https://cloud.google.com/solutions/jenkins-on-kubernetes-engine-tutorial) for setting up 
+Jenkins on GKE.
+
+<!--
+TODO(stephenashank): Add another step for the CasC config when we are able to pre-load credentials
+through the helm chart after refactoring the Google Oauth Plugin.
+-->
+Before running helm install, add the following modification to the file `jenkins/values.yaml`:
+
+* Directly below the line which reads `Agent:`, add these three lines:
+    ```yaml
+    Agent:
+        Image: us.gcr.io/jenkins-gke-plugin/jenkinsagent
+        ImageTag: latest
+        Privileged: true
+    ```
+
+The Privileged flag is needed for installing the image on the Kubernetes pods, but the agent will 
```

Comment:
Can you elaborate on this? What do you mean by "installing the image on the Kubernetes pods"?

Reply:
Yeah that was unclear. Basically I meant spinning up the agent pods with the image that we specified. I'm going through the new setup process again without the Privileged flag to make sure this is actually required. From what I remember the first time I was doing this, is that the build wouldn't start because the pod never reached a good state because the root user is used for the Docker setup which installs kubectl.

Reply:
Thanks for calling this out, turns out that flag is not required. I was conflating different errors I encountered the first time around and continuing to use this because it worked without verifying later that it was required.

Reply:
While the builds suceeded, the agent pods from this first run ended up with status: `Error with exit code 255`. Here are the stackdriver logs from that:

```
E  INFO: Protocol JNLP4-connect encountered an unexpected exception
 
E  java.util.concurrent.ExecutionException: org.jenkinsci.remoting.protocol.impl.ConnectionRefusalException: Unknown client name: default-47vzd
	at org.jenkinsci.remoting.util.SettableFuture.get(SettableFuture.java:223)
	at hudson.remoting.Engine.innerRun(Engine.java:614)
	at hudson.remoting.Engine.run(Engine.java:474)
Caused by: org.jenkinsci.remoting.protocol.impl.ConnectionRefusalException: Unknown client name: default-47vzd
	at org.jenkinsci.remoting.protocol.impl.ConnectionHeadersFilterLayer.newAbortCause(ConnectionHeadersFilterLayer.java:378)
	at org.jenkinsci.remoting.protocol.impl.ConnectionHeadersFilterLayer.onRecvClosed(ConnectionHeadersFilterLayer.java:433)
	at org.jenkinsci.remoting.protocol.ProtocolStack$Ptr.onRecvClosed(ProtocolStack.java:816)
	at org.jenkinsci.remoting.protocol.FilterLayer.onRecvClosed(FilterLayer.java:287)
	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.onRecvClosed(SSLEngineFilterLayer.java:172)
	at org.jenkinsci.remoting.protocol.ProtocolStack$Ptr.onRecvClosed(ProtocolStack.java:816)
	at org.jenkinsci.remoting.protocol.NetworkLayer.onRecvClosed(NetworkLayer.java:154)
	at org.jenkinsci.remoting.protocol.impl.BIONetworkLayer.access$1500(BIONetworkLayer.java:48)
	at org.jenkinsci.remoting.protocol.impl.BIONetworkLayer$Reader.run(BIONetworkLayer.java:247)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at hudson.remoting.Engine$1.lambda$newThread$0(Engine.java:93)
	at java.lang.Thread.run(Thread.java:748)
	Suppressed: java.nio.channels.ClosedChannelException
		... 7 more
 
E  
 
E  Mar 14, 2019 7:51:35 PM hudson.remoting.jnlp.Main$CuiListener status
 
E  INFO: Connecting to cd-jenkins-agent:50000
 
E  Mar 14, 2019 7:51:35 PM hudson.remoting.jnlp.Main$CuiListener status
 
E  INFO: Server reports protocol JNLP4-plaintext not supported, skipping
 
E  Mar 14, 2019 7:51:35 PM hudson.remoting.jnlp.Main$CuiListener status
 
E  INFO: Protocol JNLP3-connect is not enabled, skipping
 
E  Mar 14, 2019 7:51:35 PM hudson.remoting.jnlp.Main$CuiListener status
 
E  INFO: Server reports protocol JNLP2-connect not supported, skipping
 
E  Mar 14, 2019 7:51:35 PM hudson.remoting.jnlp.Main$CuiListener status
 
E  INFO: Server reports protocol JNLP-connect not supported, skipping
 
E  Mar 14, 2019 7:51:35 PM hudson.remoting.jnlp.Main$CuiListener error
 
E  SEVERE: The server rejected the connection: None of the protocols were accepted
 
E  java.lang.Exception: The server rejected the connection: None of the protocols were accepted
	at hudson.remoting.Engine.onConnectionRejected(Engine.java:682)
	at hudson.remoting.Engine.innerRun(Engine.java:639)
	at hudson.remoting.Engine.run(Engine.java:474)

```

Removing the `--version 0.16.6` from the helm install seems to have fixed that though.